the-collab-lab/tcl-40-smart-shopping-list
# Added a welcome message for user if no items in list, and light styling

## Description
- Added conditional rendering after checking to see if `data` is an empty array:
-- If there are items in the list, the items will display automatically, if there are no items in the list, a welcome message and a CTA (link styled as a button) are displayed.

- Added styles to Footer.css and App.css
- Created List.css file; added styles to links and created layout per wireframe.
- Added `react-icons` for 'logo' 
- Added individual footer component to both the list view and addItems view so that you can toggle between each view.



## Related Issue

Closes #7


## Acceptance Criteria

- The list view, when there are no items to display, should show a prompt (e.g., a button) for the user to add their first item

Notes:
- The wireframe is a little short on the "welcoming" part of the story, but otherwise is functionally complete for the purposes of the AC. https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/list-empty.png


## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|   ✓ | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before

<img width="369" alt="Screen Shot 2022-04-27 at 6 10 35 PM" src="https://user-images.githubusercontent.com/52385888/165639578-e66cb7ad-b08c-47ff-96d9-292e21a80bb3.png">


### After
<img width="426" alt="Screen Shot 2022-04-27 at 6 09 43 PM" src="https://user-images.githubusercontent.com/52385888/165639508-884efcf7-a847-4146-a687-b941f5e11576.png">






## Testing Steps / QA Criteria

1. run `npm install` to add `react-icons`
2. run `npm start`
3. Ensure you do not have a token in your `localStorage` or confirm you are using a token (joining a list) that does not contain any items.
4. If you are creating a new list, you should see the new UI (welcome message and CTA). 
5. If you are joining a list that has items, you will just see the items listed. They will appear at the top of the screen and you will see the newly styled links (List || Add Item) at the bottom of the screen.  
6.  You are also able to toggle between the list view and the add Item view.
